File: src/main/java/com/freya02/botcommands/internal/application/context/ContextCommandParameter.java
package com.freya02.botcommands.internal.application.context;

import com.freya02.botcommands.api.BContext;
import com.freya02.botcommands.api.application.CommandPath;
import com.freya02.botcommands.internal.application.ApplicationCommandParameter;

import java.lang.reflect.Parameter;

public class ContextCommandParameter<T> extends ApplicationCommandParameter<T> {
	public ContextCommandParameter(BContext context, CommandPath path, Class<T> resolverType, Parameter parameter, int index) {
		super(context, path, resolverType, parameter, index);
	}
}

File: src/main/kotlin/com/freya02/botcommands/internal/application/context/ContextCommandParameter.kt
package com.freya02.botcommands.internal.application.context

import com.freya02.botcommands.internal.parameters.MethodParameter
import com.freya02.botcommands.internal.parameters.MethodParameterType
import kotlin.reflect.KParameter

abstract class ContextCommandParameter<R>(
    parameter: KParameter,
    val resolver: R
) : MethodParameter {
    override val methodParameterType = MethodParameterType.OPTION
    override val kParameter = parameter
}